simulator/simple_dds/dds_service.py

```diff
@@ -0,0 +1,214 @@
+import logging
+from threading import Lock
+from singleton import Singleton
+from simple_dds import entity
+
+# Possíveis problemas de uso mútuo nesta classe? Veremos quando testarmos.
+class UniqueHandleController(metaclass=Singleton):
+
+    def __init__(self):
+        self.next_available_handle = 1
+        self.lock = Lock()
+    
+    def generate_handle(self):
+        handle = None
+        with self.lock:
+            handle = self.next_available_handle
+            self.next_available_handle += 1
+        return handle
+
+# Classe criada para impedir conflito entre leitura e escrita.
+# Poderia fazer a lógica usando um Resource do simpy, mas prefiro evitar o uso..
+# ... do simpy na lógica do DDS.
+class Data_Buffer:
+
+    def __init__(self):
+        self.data_buffer = []
+        self.lock = Lock()
+
+    def write_to(self, data):
+        with self.lock:
+            self.data_buffer.append(data)
+
+    def read_from(self, container):
+        with self.lock:
+            container = copy(self.data_buffer)
+            self.data_buffer = []
+
+    def is_empty(self):
+        return len(self.data_buffer) == 0
+
+# Talvez seja necessário fazer dictionaries personalizados, que evitem problemas de acesso mútuo.
+class DDS_Service(entity.Entity):
+
+    def __init__(self, driver):
+        self.handle_controller = UniqueHandleController()
+        self.instance_handle = self.handle_controller.generate_handle()
+        self.driver = driver
+        self.peer_list = []
+        self.handles = {} # Handle: Entity
+        self.participants = {} # Handle: Participant
+        self.local_participants = {}
+        self.topics = {}  # Topic name: Topic
+        self.data_objects = {} # Handle: Data object
+        self.message_handlers = {}
+
+        self.add_message_handler_methods()
+        self.attach_msg_reception_handler_to_driver()
+        self.discover_peers()
+        self.request_full_domain_data()
+
+    def set_instance_handle(self, handle):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/gh/lesunb/p2psimpy/pullRequest?prid=5588062)

Reply:
Not in this case. This is an implementation of an abstract method, and, for this class, it's not supposed to interact or alter attributes. Instead, it raises an error, as we don't want it to ever be called.